src/mmwtex.c

```diff
@@ -3047,55 +3057,94 @@ void printTexLongMath(nmbrString *mathString,
-            " && ",
-
-            // Local label if any e.g. "@2:"
-            (htmLocLab[0] != 0) ? cat(htmLocLab, "\\ ", NULL) : "",
-
-            " & ",
-            texLine,
-            // Don't put space to help prevent bad line break
-
-            // Surround \ref with \mbox for non-math-mode
-            // symbolic labels (due to \tag{..} in mmcmds.c).  Also,
-            // move hypotheses to after referenced label.
-            "&",
-            "(",
-
-            // Don't make local label a \ref
-            (htmRef[0] != '@') ?
-                cat("\\mbox{\\ref{eq:", htmRef, "}}", NULL)
-                : htmRef,
-
-            htmHyp[0] ? "," : "",
-            htmHyp,
-            ")\\notag", NULL),
-
-            "    \\notag \\\\ && & \\qquad ", // Continuation line prefix
-            " ");
+        texFull = cat("  ", // texFull[0] should not be a "{" character.
+          // If not first step, so print "\\" LaTeX line break
+          !strcmp(htmStep, "1") ? "" : "\\\\ ",
+          htmStep, // Step number
+          " && ",
+
+          // Local label if any e.g. "@2:"
+          (htmLocLab[0] != 0) ? cat(htmLocLab, "\\ ", NULL) : "",
+
+          " & ",
+          texLine,
+          // Don't put space to help prevent bad line break
+
+          // Surround \ref with \mbox for non-math-mode
+          // symbolic labels (due to \tag{..} in mmcmds.c).  Also,
+          // move hypotheses to after referenced label.
+          "&",
+          "(",
+
+          // Don't make local label a \ref
+          (htmRef[0] != '@') ?
+              cat("\\mbox{\\ref{eq:", htmRef, "}}", NULL)
+              : htmRef,
+
+          htmHyp[0] ? "," : "",
+          htmHyp,
+          ")\\notag", NULL),
+
+        // To avoid generating incorrect TeX, line breaking is forbidden inside
+        // scopes of curly braces.  However breaking between '\{' and '\}' is
+        // allowed.
+        // The spaces that should not be matched with 'breakMatch' are
+        // temporarily changed to ASCII 3 before the 'printLongLine' procedure
+        // is called.  The procedure rewraps the 'line' argument matching the
+        // unchanged spaces only, thus ensuring bad breaks will be avoided.
+        // The reverse is done in the 'print2()' function, where all ASCII 3's
+        // are converted back to spaces.
+        // k counts the scope level we are in.
+        k = 0;
+        n = (long)strlen(texFull);
+        for (j = 1; j < n; j++) { // We don't need to check texFull[0].
+          // We enter a non "\{" scope.
+          if (texFull[j] == '{' && texFull[j - 1] != '\\') k++;
+          // We escape a non "\}" scope.
+          if (texFull[j] == '}' && texFull[j - 1] != '\\') k--;
+          // If k > 0 then we are inside a scope.
+          if (texFull[j] == ' ' && k > 0) texFull[j] = QUOTED_SPACE;
+        }
+        printLongLine(texFull, "    \\notag \\\\ && & \\qquad ", /* Continuation line prefix */ " ");
```

Comment:
Looks like a lot of duplication with the previous part.
Then again the duplicated code was already there before.

Reply:
I unduplicated the brace count and the `printLongLine` call in a new commit.

Reply:
Looks good!